routing/router.go

```diff
@@ -1822,19 +1822,31 @@ func (r *ChannelRouter) sendPayment(
 func (r *ChannelRouter) processSendError(paySession PaymentSession,
 	rt *route.Route, fErr *htlcswitch.ForwardingError) bool {
 
-	errSource := fErr.ErrorSource
-	errVertex := route.NewVertex(errSource)
+	failureSourceIdx := fErr.FailureSourceIdx
+	log.Tracef("node=%x reported failure when sending htlc", failureSourceIdx)
```

Comment:
IMO we should still log the pubkey as it makes it easier to obtain information from the logs. Otherwise, with only the index you'll need to  scroll back up many lines to count the nodes in the route, but even then the pubkey may be logged in a `btcec.PublicKey` only exposing the raw `big.Int`.

Reply:
Agreed and added.